### PR TITLE
Use StateResolutionHandler to resolve state in persist_events

### PR DESCRIPTION
**Pull request by richvdh**
... and thus benefit (hopefully) from its cache.

### CODE REVIEW

File: synapse/storage/events.py [hunk 1]
```diff
@@ -515,16 +515,21 @@ def _get_new_state_after_events(self, events_context, new_latest_event_ids):
                     if ctx.current_state_ids is None:
                         raise Exception("Unknown current state")
 
+                    if ctx.state_group is None:
+                        # I don't think this can happen, but let's double-check
+                        raise Exception(
+                            "Context for new extremity event %s has no state "
+                            "group" % event_id,
```
Comment (erikjohnston) [inline synapse/storage/events.py]: Nit: I do prefer always using tuples rather than relying on strings working.

File: synapse/storage/events.py [hunk 2]
```diff
@@ -586,6 +586,7 @@ def get_events(ev_ids):
 
                 current_state = yield resolve_events_with_factory(
                     state_sets,
+                    event_map={},
```
Comment (erikjohnston) [inline synapse/storage/events.py]: Should this be None to be a bit more consistent?

File: synapse/storage/events.py (hunk 1 above)
Comment (richvdh) [reply synapse/storage/events.py]: as in you'd prefer `(event_id, )` ?

File: synapse/storage/events.py (hunk 2 above)
Comment (richvdh) [reply synapse/storage/events.py]: plausibly, but it's replaced in https://github.com/matrix-org/synapse/pull/2864/commits/ebfe64e3d69d0047ee9902a05beaf0249f11e072 anyway

File: synapse/storage/events.py (hunk 1 above)
Comment (erikjohnston) [reply synapse/storage/events.py]: Generally yes, (or `(event_id,)` without the space), as its very confusing if there is a bug and `event_id` suddenly becomes iterable and then the logging explodes. In this particular case its probably fine though

File: synapse/storage/events.py (hunk 2 above)
Comment (erikjohnston) [reply synapse/storage/events.py]: Yup, didn't realise until after i had submitted the review

File: synapse/storage/events.py (hunk 1 above)
Comment (richvdh) [reply synapse/storage/events.py]: fair

**Pull request merged**